haddock-library/src/Documentation/Haddock/Parser/Monad.hs

```diff
@@ -73,6 +75,25 @@ char = lift . Attoparsec.char
 char8 :: Char -> Parser Word8
 char8 = lift . Attoparsec.char8
 
+-- | Peek a unicode character and return the number of bytes that it took up
+peekUnicode :: Parser (Char, Int)
+peekUnicode = lift $ Attoparsec.lookAhead $ do
+  
+  -- attoparsec's take fails on shorter inputs rather than truncate
+  bs <- Attoparsec.choice (map Attoparsec.take [4,3,2,1])
+  
+  let c = head . decodeUtf8 $ bs
```

Comment:
This is all so horrible. But I agree with you that it is the simplest thing possible. Once GHC ships `text` as boot library we could think about rewriting the parser to avoid these hacks.

Reply:
I agree this is madness. Using `Data.Attoparsec.Text` would be so much cleaner.

Reply:
On a related note: Changing the types from `String` to `Text` will have an impact on interface size for haddock. Currently it always stores `Char` as 4 byte. Using a more compact encoding like utf-8 will reduce interface size immensely.